src/LayerEditor/ui/panels/diagram.py

```diff
@@ -106,38 +106,60 @@ def __init__(self, parent=None):
   
         self.set_data()    
         self.setSceneRect(0, 0, 20, 20)
-        
+
+    def remove_graphical_object(self, obj):
+        self.removeItem(obj)
+        self.regionsUpdateRequired.emit()
+
+    def add_graphical_object(self, obj):
+        self.addItem(obj)
+        dim = None
+        layer_id = cfg.diagram.regions.current_layer_id
+        if isinstance(obj, Point):
+            dim = 0
+        elif isinstance(obj, Line):
+            dim = 1
+        elif isinstance(obj, Polygon):
+            dim = 2
+        if dim is not None:
+            reg_id = cfg.main_window.regions.get_current_region()
+            if not reg_id == 0:
+                region = cfg.diagram.regions.regions[reg_id]
+                pas = region.cmp_shape_dim(layer_id, dim)
+                if pas:
+                    self.regionsUpdateRequired.emit()
```

Comment:
Explain why we need update for this case.

Reply:
The whole function can be reduced in the way of complementary function remove_graphical_object(). The whole segment only filters unnecessary updates of the region panel.
The main purpose of the update signal on object addition is disabling the remove button once the region is used.